Dev

Proxify opened the pull request.

ccarpenter04 commented on src/main/java/net/botwithus/api/game/hud/prayer/Prayer.java:
```diff
@@ -46,6 +46,13 @@ public static int getMaxPrayerPoints() {
+     *
+     * @return the player's prayer book (0 = Normal, 1 = Curses)
+     */
+    public static int getPrayerBook() { return VarManager.getVarbitValue(16789); }
```
Instead of relying on magic numbers like 16789, define a constant with a proper name that is easily distinguishable.
Then, instead of having it return an int, have it return an enum value of type PrayerBook or similar.

ccarpenter04 replied:
I'd personally just annotate it @Nullable and make it return null if it can't resolve the book for some reason, however if you want to take that approach instead you should do either PrayerBook.UNKNOWN, UNDEFINED, or an _

Also, since it's in the Prayer class, the naming is a bit redundant. Prayer.getPrayerBook() -> Prayer.getCurrentBook/getCurrentPrayerBook if you want to be verbose but something should indicate that it's not getting "THE" prayer book but rather the current one. And of course adjust JDocs as necessary.

Anjato replied:
defaulted to a null return value and renamed the method

ccarpenter04 reviewed (changes requested):
Quick changes

ccarpenter04 commented on src/main/java/net/botwithus/api/game/hud/prayer/Prayer.java:
```diff
@@ -8,6 +8,12 @@ public final class Prayer {
+        NULL,
+        Normal,
+        Curses
+    }
```
Enum naming conventions require the members to be all caps, so NORMAL and CURSES

Anjato replied:
fixed enums

ccarpenter04 commented on src/main/java/net/botwithus/api/game/hud/prayer/Prayer.java:
```diff
@@ -53,17 +52,17 @@ public static int getMaxPrayerPoints() {
-            default -> PrayerBook.NULL;
+            case 0 -> PrayerBook.NORMAL;
+            case 1 -> PrayerBook.CURSES;
+            default -> null;
```
My only comment here is if null is going to be returned, we should probably log the value that was returned by the VarManager.

Anjato replied:
where should the value be logged? was trying to find something in other methods but I'm not seeing anything else logging similar situations

Anjato replied:
@ccarpenter04 bumping this in case it got missed

ccarpenter04 replied:
Team, how we should have the ex-api logging system setup?

Proxify committed:
Merge pull request #5 from BotWithUs/master

Merge pull request #4 from BotWithUs/dev

Proxify committed:
Updates to LootInventory class

- Added the following methods to LootInventory class.
    - #isEnabled
    - #isMultipleItemsOpenEnabled
    - #isAreaLootEnabled
    - #isDecideByMonetaryValueEnabled
    - #getMonetaryValue

Proxify committed:
Added LootInventory#open

CodeByLM committed:
Update LootInventory.java

Fixed varbit typo and added getMonetaryValue() enum

CodeByLM committed:
Added Divine Rage and Eclipsed Soul prayer

CodeByLM committed:
Merge pull request #1 from louis-dnb/master

Added Divine Rage and Eclipsed Soul prayer

Proxify committed:
Merge pull request #7 from louis-dnb/dev

Update LootInventory.java
